src/app/shared/api-locator.service.ts

```diff
@@ -33,27 +34,36 @@ export class ApiLocatorService {
   }
 
   get witApiUrl(): string {
-    return this.buildApiUrl('wit');
+    return this.buildApiUrl('wit', (config) => config.witApiUrl);
   }
 
   get forgeApiUrl(): string {
-    let tmp=this.buildApiUrl('forge');
+    let tmp=this.buildApiUrl('forge', (config) => config.forgeApiUrl);
     return tmp
   }
 
   get ssoApiUrl(): string {
-    return this.buildApiUrl('sso');
+    return this.buildApiUrl('sso', (config) => config.ssoApiUrl);
   }
 
   get recommenderApiUrl(): string {
-    return this.buildApiUrl('recommender');
+    return this.buildApiUrl('recommender', (config) => config.recommenderApiUrl);
   }
 
   private loadEnvVar(key: string): void {
     this.envVars.set(key, process.env[this.DEFAULT_API_ENV_VAR_NAMES.get(key)]);
   }
 
-  private buildApiUrl(key: string): string {
+  private buildApiUrl(key: string, fn: (OAuthConfig) => string): string {
```

Comment:
I *think* it should be ok, as in prod/prod-preview we don't use env vars for these, just relative vars. This might well break dev, but that is more easy to test. Given our goal right now is to remove tech debt I would prefer to break prod-preview or even prod and get this right ;-)